tests/sentry/workflow_engine/migration_helpers/test_migrate_alert_rule.py

```diff
@@ -409,209 +308,34 @@ def test_update_metric_alert(self):
         assert detector_state.state == str(DetectorPriorityLevel.OK.value)
         assert detector_state.active is False
 
-    def test_update_metric_alert_owner(self):
-        migrate_alert_rule(self.metric_alert, self.rpc_user)
+    def test_dual_update_metric_alert_owner(self):
+        detector = self.detector
         updated_fields = {
             "user_id": self.user.id,
             "team_id": None,
         }
-        alert_rule_detector = AlertRuleDetector.objects.get(alert_rule=self.metric_alert)
-        detector = alert_rule_detector.detector
-        assert detector.owner_user_id == self.metric_alert.user_id
-        assert detector.owner_team_id == self.metric_alert.team_id
 
-        update_migrated_alert_rule(self.metric_alert, updated_fields)
+        dual_update_migrated_alert_rule(self.metric_alert, updated_fields)
         detector.refresh_from_db()
 
         assert detector.owner_user_id == self.user.id
         assert detector.owner_team_id is None
 
     def test_update_metric_alert_config(self):
-        migrate_alert_rule(self.metric_alert, self.rpc_user)
+        detector = self.detector
```

Comment:
nit; i'd just use `self.detector` instead of `detector` later; that way it's clear you're using a variable from the class instance